pilot/pkg/config/kube/gateway/deploymentcontroller.go

```diff
@@ -250,6 +281,16 @@ func (d *DeploymentController) configureIstioGateway(log *istiolog.Scope, gw gat
 	return nil
 }
 
+func (d *DeploymentController) RenderServiceAccountApply(input MergedInput) *corev1ac.ServiceAccountApplyConfiguration {
+	return corev1ac.ServiceAccount(input.ServiceAccount, input.Namespace).
+		WithLabels(map[string]string{GatewayNameLabel: input.Name}).
```

Comment:
we will want the ManagedByControllerLabel label as well I think. We may want to move this to using `ApplyTemplate` for consistency (it adds that) as well maybe